core/create/src/index.ts

```diff
@@ -348,8 +348,10 @@ async function optionsPrompt(config: Config): Promise<boolean> {
       continue
     }
 
-    const [optional, required] = partition(Object.entries(options), ([, optionType]) =>
-      optionType.endsWith('?')
+    const [optional, required] = partition(
+      Object.entries(options),
+      (schema): schema is [string, `${SchemaType}?`] =>
```

Comment:
We know that the `schema[1]` must be optional (if the type guard succeeds) as we check that the schema ends with a question mark: `schema[1].endsWith('?')`.

Reply:
can I just confirm that ```(schema: [string, (`${SchemaType}?`| SchemaType) ])``` is the same type guard as yours for my own understanding?

Reply:
No, a string in the `SchemaType` type cannot end with a `?` (because of how the type is defined,) whereas a string in ``` `${SchemaType}?` ``` _must_ end in a `?` because it's explicitly in the type literal. So ``` `${SchemaType}?`| SchemaType``` can include the strings `number` or `number?` whereas `SchemaType` could only include the string `number` and ``` `${SchemaType}?` ``` could only include the string `number?`. The point of the type guard is to only filter through words that end in `?` and therefore are in the type ``` `${SchemaType}?` ``` but not `SchemaType` (and now, additionally in this PR, are not a function.)

Reply:
Thanks for explaining, I understand that the point of the type guard is to only filter through words that end in `?` and therefore are in the type `${SchemaType}?`, but how come this
```
(schema: [string, (`${SchemaType}?`| SchemaType) ]) =>
        typeof schema[1] === 'string' && schema[1].endsWith('?')
```
is invalid? since we need to filter through words that end in `?`  that means schema[1] could be either `${SchemaType}?`or `SchemaType`. 

As for the type guard you have, 

```
(schema): schema is [string, `${SchemaType}?`] =>
        typeof schema[1] === 'string' && schema[1].endsWith('?')
```

schema[1] is `${SchemaType}?` so it's a bit strange to me that we are filtering for words that end in  `?` when we already know that it is schema[1] must be optional. Am I missing something?

Reply:
Ah yes, sorry, I understand what you mean now. Yes, at the moment the `schema` argument's type is implicit but if we were to write out the full function signature the type would be
```ts
(schema: [string, (`${SchemaType}?` | SchemaType)]): schema is [string, `${SchemaType}?`]
```

Reply:
oh right thanks, I understand what you mean now - I misinterpreted the function return type as the `schema` parameter type 😆

Reply:
and I misinterpreted your parameter type as the return type 🥴